src/modules/ekf2/EKF/mag_control.cpp

```diff
@@ -303,7 +311,7 @@ void Ekf::checkYawAngleObservability()
 				: _accel_lpf_NE.norm() > 2.0f * _params.mag_acc_gate;
 
 	_yaw_angle_observable = _yaw_angle_observable
-				&& (_control_status.flags.gps || _control_status.flags.ev_pos); // Do we have to add ev_vel here?
+				&& (_control_status.flags.gps || (_control_status.flags.ev_pos && _control_status.flags.yaw_align));
```

Comment:
The yaw angle is only observable if the ev_pos data is in the local frame, if the data is in body frame and that the yaw is aligned using the mag, the yaw isn't observable.

Reply:
Yes that's the intent here, and you're obviously correct about the potential problem, but I believe it's not really an issue due to how intertwined the control logic is between EV yaw/pos <=> mag <=> GPS. This is related to the scattered/distributed state machine issues I mentioned a few months ago.

We *should* only have ev_pos + yaw_align if the incoming EV is NED (or at least tells us it is...), however I agree from these flags along it's a little ambiguous.

Assuming EV data is an arbitrary world heading (FRD)
 - if GPS is bad/inactive
   - EV yaw is allowed to start if GPS isn't active (_control_status.flags.yaw_align = false)
   - EV yaw will stop all mag fusion
   - EV position fusion using data directly (no bias estimate, no rotation, etc) 
 - GPS good/active
   -  EV yaw continuing conditions no longer passing
   - mag fusion no longer blocked
   - EV position rotated into EKF frame and position bias estimator activated

Let me think about a proper way to capture and enforce this in the code.